component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -675,14 +702,12 @@ private void processEmailOTPMandatory(AuthenticationContext context, HttpServlet
      * @throws AuthenticationFailedException
      */
     private void updateEmailAddressForUsername(AuthenticationContext context, String username)
-            throws AuthenticationFailedException {
+            throws AuthenticationFailedException, UserStoreException {
         String tenantDomain = context.getTenantDomain();
-        if (username != null && !context.isRetrying()) {
-            Map<String, String> attributes = new HashMap<>();
-            attributes.put(EmailOTPAuthenticatorConstants.EMAIL_CLAIM,
-                    String.valueOf(context.getProperty(EmailOTPAuthenticatorConstants.REQUESTED_USER_EMAIL)));
-            updateUserAttribute(MultitenantUtils.getTenantAwareUsername(username), attributes, tenantDomain);
-        }
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put(EmailOTPAuthenticatorConstants.EMAIL_CLAIM,
+                String.valueOf(context.getProperty(EmailOTPAuthenticatorConstants.REQUESTED_USER_EMAIL)));
+        updateUserAttribute(MultitenantUtils.getTenantAwareUsername(username), attributes, tenantDomain);
```

Comment:
verify the scenario works as expected with a SaaS application

Reply:
This flow is verified an fixed an issue